### PR TITLE
Fix Alt+number shortcuts for tabs in chat overlay

### LINKED ISSUE #7608: Alt+[number] combination, in Chat Overlay, starts from number 2
**Describe the bug:** (Use the image below as the example for this explanation)

When I press <kbd>Alt</kbd>+<kbd>2</kbd>, it opens `#announce`; <kbd>Alt</kbd>+<kbd>3</kbd> opens `#italian`, etc.
So it seems like it's skipping the number 1.

**Screenshots or videos showing encountered issue:**
chat

**osu!lazer version:** 2020.118.0

### PR DESCRIPTION
**Pull request by bdach**
Resolves #7608.

- [x] Depends on #7683 (didn't have to, I just wanted to get the inevitable merge conflict out of the way).

# Summary

Add test for <kbd>Alt</kbd>+[<kbd>0</kbd>-<kbd>9</kbd>] key shortcuts and resolve the regression by checking the channel type at the point of enumerating channels.

# Remarks

Since I didn't want to recreate the entire container for a single test, I just made the pre-existing tests take in 10 channels instead of the previous 3 and adjusted data-dependent logic appropriately. I'm aware that this introduces interdependence to the test cases, but I don't think it's that bad. I can separate them out if that's deemed preferable.

### CODE REVIEW

**Comment by swoolcock** [comment]
Tab-switching on macOS is generally done with cmd+number, so this should be a `PlatformAction`, or accept both.

**Comment by bdach** [comment]
That's fair, it's definitely something I can follow up on, it might be better suited as a separate PR, though. (And I'd appreciate help testing it as a Mac-less person!)

File: osu.Game/Overlays/ChatOverlay.cs [hunk 1]
```diff
@@ -321,8 +321,11 @@ protected override void OnDragEnd(DragEndEvent e)
 
         private void selectTab(int index)
         {
-            var channel = ChannelTabControl.Items.Skip(index).FirstOrDefault();
-            if (channel != null && !(channel is ChannelSelectorTabItem.ChannelSelectorTabChannel))
+            var channel = ChannelTabControl.Items
+                                           .Where(tab => !(tab is ChannelSelectorTabItem.ChannelSelectorTabChannel))
+                                           .Skip(index)
+                                           .FirstOrDefault();
```
Comment (huoyaoyuan) [inline osu.Game/Overlays/ChatOverlay.cs]: `ElementAtOrDefault` should be enough

Comment (peppy) [reply osu.Game/Overlays/ChatOverlay.cs]: wow, that exists?!

Comment (huoyaoyuan) [reply osu.Game/Overlays/ChatOverlay.cs]: according to the document, it existed from the start of LINQ

**Review by peppy — APPROVED** [review]

**Comment by bdach** [comment]
As for the suggestion to switch to <kbd>Cmd</kbd>/<kbd>Ctrl</kbd>+[<kbd>0</kbd>-<kbd>9</kbd>] - I just remembered that <kbd>Ctrl</kbd>+[<kbd>1</kbd>-<kbd>4</kbd>] are already used for switching rulesets in the top bar, so I'm going to leave this as-is for now to avoid conflicts.

**Pull request merged**